parsing/parchmint-parser.js

```diff
@@ -202,6 +218,44 @@ class ParchmintParser {
         });
     }
 
+    /**
+     * Parse a JSON object for the Connection Features.
+     *
+     * Fills the connFeatures map with Connection Features where the keys are
+     * Connection IDs, and the values are Arrays of Connection Segment objects
+     * that correspond to that Connection. Sets the parser invalid if duplicate
+     * IDs are found.
+     *
+     * @since 1.0.0
+     *
+     * @see connFeatures
+     *
+     * @param {object}  jsonObj A parsed JSON object representing the Parchmint
+     *                          file.
+     */
+    parseConnectionFeatures(jsonObj) {
+        jsonObj['features'].forEach((value, index) => {
+            // First check whether this is a Connection Feature
+            if (value['type']) {
+                // Next check that the ID of this segment is unique
+                if (this.isUniqueID(value['id'])) {
+                    // Finally add the Connection Feature to the map
+                    let tempFeat = new ConnectionSegment(value['name'], value['id'], value['width'], value['depth'],
+                            ParchmintParser.parseCoord(value['source']), ParchmintParser.parseCoord(value['sink']));
+                    if (this.connFeatures.has(value['connection'])) {
+                        this.connFeatures.get(value['connection']).push(tempFeat);
+                    } else {
+                        this.connFeatures.set(value['connection'], [tempFeat]);
+                    }
+                } else {
+                    this.valid = false;
+                    console.log('Parser: Duplicate IDs (' + value['id'] + ') exist for the Connection Features list.' +
+                            ' Skipping Connection Feature with name "' + value['name'] + '" at index ' + index + '.');
+                }
+            }
+        });
+    }
```

Comment:
I know its not really necessary since the objects get added where necessary in the function but its a bit strange that some of the functions return their elements and some don't. Two options we have here (if we think it needs to be fixed at all) is to return the array of elements anyway even though they are (currently) unused. Another, which would probably be more difficult, would be to return the array of elements (or a single element) and have another function which adds it where necessary (separation of concerns)

Reply:
My thinking was that the top level key functions fill the maps and arrays, where ```parsePort``` and ```parseTerminal``` are more like helper functions to be used in the top level key functions. I had originally thought about writing separate functions to parse a single element, but because they would only be used in the current functions I decided to write them inline.

I'll return a reference to the finished map/array for each key if you think that would be useful.

Reply:
I can see the benefits down the line if we want to use these functions in a different way (as we can then extract the elements) but what might be more expedient if you don't want to or don't think we should do that now would be to choose a different naming convention for the ones that return (perhaps `getParsedPort`, etc.).

Reply:
That's a good idea I'll rename the functions that return, as well as write ```getParsed*``` functions for each of the top level keys.